### PR TITLE
Python 3 enabled, fixed tar file name generation and subporcess outpu…

### PR DESCRIPTION
**Pull request by chrisAtRedis**
…t encoding

### CODE REVIEW

File: log_collector.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Comment (RoeyPrat) [inline log_collector.py]: sadly, not everyone is using python3

File: log_collector.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 from collections import OrderedDict
 import shutil
 import json
+import pathlib
```
Comment (RoeyPrat) [inline log_collector.py]: We would prefer to not install any libraries, as this requires the user to install them as well.
If this script becomes more complex to maintain, it is no longer a simple script, and should be deployed using containers.

File: log_collector.py (hunk 1 above)
Comment (chrisAtRedis) [reply log_collector.py]: One of the major points of this PR (and subsequent work) was to get this code Python 3 ready.

File: log_collector.py (hunk 2 above)
Comment (chrisAtRedis) [reply log_collector.py]: pathlib is part of the standard Python 3 library (cf. https://docs.python.org/3/library/pathlib.html), so there's no need for a user to explicitly install this module.  In contrast to yaml :-).

File: log_collector.py (hunk 1 above)
Comment (ushachar) [reply log_collector.py]: Not sure we should assume python3 is available on all instances which run this script.

**Comment by danni-m** [comment]
@chrisAtRedis now that #64 is merged, can you take a look to see if it works for you?

**Comment by chrisAtRedis** [comment]
Hi Danni,
File name generation in the tgz is still broken. Please take a look at the revised archive_files implementation (branch fixed_log_collector in https://github.com/chrisAtRedis/redis-enterprise-k8s-docs/blob/fixed_log_collector/log_collector.py) which now works on Python 2 as the pathlib requirement has bee removed.

**Comment by danni-m** [comment]
@chrisAtRedis we tested it on both Python version, can you paste the stacktrace here?

**Comment by chrisAtRedis** [comment]
No stacktrace. 

The file name generation in the tar archive is broken:

tar tvf redis_enterprise_k8s_debug_info_20190903-174713.tar.gz 
drwxr-xr-x  0 christoph staff       0 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/
-rw-r--r--  0 christoph staff    6623 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/Deployment.yaml
-rw-r--r--  0 christoph staff    2709 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/Services.yaml
-rw-r--r--  0 christoph staff    1680 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/resources_list
-rw-r--r--  0 christoph staff      83 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/Ingress.yaml
-rw-r--r--  0 christoph staff    2042 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/RedisEnterpriseCluster.yaml
drwxr-xr-x  0 christoph staff       0 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/pods/
-rw-r--r--  0 christoph staff    9112 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/pods/redis-enterprise-2.log
-rw-r--r--  0 christoph staff    9424 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/pods/redis-enterprise-1.log
-rw-r--r--  0 christoph staff  220637 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/pods/redis-enterprise-operator-8475fc7f44-s7kfq.log
-rw-r--r--  0 christoph staff    8533 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/pods/redis-enterprise-0.log
-rw-r--r--  0 christoph staff 3670441 Sep  3 17:48 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/pods/redis-enterprise-services-rigger-5b9dc56d5f-gkpk8.log
-rw-r--r--  0 christoph staff      83 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/ConfigMap.yaml
-rw-r--r--  0 christoph staff    5312 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/StatefulSet.yaml
-rw-r--r--  0 christoph staff     686 Sep  3 17:47 redis_enterprise_k8s_debug_info_20190903-174713.tar.gz/cluster_info

When you try to unpack the archive, tar barfs as the directory name used to unpack into is the same as the .tar.gz file.

**Comment by RoeyPrat** [comment]
@chrisAtRedis I'm closing this PR, as I think #76 addressed the issues here, without breaking python2 compat. 
When you have time, please try the new version, and let me know what you think :)

**Pull request closed without merging**